spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/DefaultReleaseManager.java

```diff
@@ -238,6 +257,70 @@ private void updateCountProperty(Map<String, Object> model, String appsCount) {
 		deploymentPropertiesMap.put(SPRING_CLOUD_DEPLOYER_COUNT, appsCount);
 	}
 
+	public Mono<Release> statusReactive(Release release) {
+		return Mono.defer(() -> {
+			if (release.getInfo().getStatus().getStatusCode().equals(StatusCode.DELETED)) {
+				return Mono.just(release);
+			}
+			AppDeployer appDeployer = this.deployerRepository.findByNameRequired(release.getPlatformName())
+					.getAppDeployer();
+			AppDeployerData appDeployerData = this.appDeployerDataRepository
+				.findByReleaseNameAndReleaseVersion(release.getName(), release.getVersion());
+			if (appDeployerData == null) {
+				logger.warn(String.format("Could not get status for release %s-v%s.  No app deployer data found.",
+					release.getName(), release.getVersion()));
+				return Mono.just(release);
+			}
+			List<String> deploymentIds = appDeployerData.getDeploymentIds();
+			logger.info("Getting status for {} using deploymentIds {}", release,
+					StringUtils.collectionToCommaDelimitedString(deploymentIds));
+
+			if (!deploymentIds.isEmpty()) {
+				Map<String, String> appNameDeploymentIdMap = appDeployerData.getDeploymentDataAsMap();
+				return Flux.fromIterable(appNameDeploymentIdMap.entrySet())
+					.flatMap(nameDeploymentId -> {
+						String deploymentId = nameDeploymentId.getValue();
+						return appDeployer.statusReactive(deploymentId);
```

Comment:
This call still is made to retrieve ApplicationDetail while we can limit this to fetch only the summary if that is all that we need.  We can split the statusReactive(Release release) into two parts 1)with ApplicationDetail 2) without ApplicationDetail.